code.rb

```diff
@@ -0,0 +1,16 @@
+class Code
+  DEST_CODE = {
+    "null" => "000",
+    "M" => "001",
+    "D" => "010",
+    "MD" => "011",
+    "A" => "100",
+    "AM" => "101",
+    "AD" => "110",
+    "AMD" => "111",
+  }
+
+  def dest(mnemonic)
+    DEST_CODE[mnemonic]
```

Comment:
Ruby note: I think it might be a little better to use `#fetch` instead of `#[]` here, since you probably want to raise an exception if the parser somehow provides in a mnemonic you don’t recognise, rather than silently returning a `nil` which might cause trouble in a different part of the program. (Same advice applies to the other methods in this PR.)

Reply:
Done!